crates/lib3h/src/engine/ghost_engine.rs

```diff
@@ -0,0 +1,26 @@
+use lib3h_ghost_actor::GhostParentWrapper;
+use lib3h_protocol::protocol::*;
+/// the context when making a request from core
+/// this is always the request_id
+pub struct CoreRequestContext(String);
```

Comment:
Prefer to rename it to `ClientRequestContext` as we are avoiding the term `Core`